### ISSUE TITLE
[ENHANCEMENT] Take advantage of the attach/detach mechanism in Longhorn v1.5+

### ISSUE THREAD

**Issue by ejweber** [harvester/harvester#4907]
While looking into a few Harvester issues, I noticed https://github.com/harvester/harvester/pull/3670 and https://github.com/harvester/harvester/issues/3648. It appears from these (though I need to collect more context) that Harvester manually attached Longhorn volumes to nodes in order to ensure snapshots/backups can be taken.

This behavior makes sense for Longhorn <= `v1.4.x`. In those versions, Longhorn did not have any mechanism to attach the volume automatically to the node and perform the operation, so these operations were generally stuck until a volume was attached. However, in v1.5.x, Longhorn added a new attach/detach controller that automatically attaches volumes for these operations and then detaches them. See https://github.com/longhorn/longhorn/blob/master/enhancements/20221024-longhorn-volumeattachment.md and https://github.com/longhorn/longhorn/issues/3726.

I don't know the full extent of Harvester's need to manipulate Longhorn volumes, but I think it may be possible to remove or reduce this functionality in Harvester. It may even be safer to do so to avoid situations in which Harvester's attach/detach logic fights Longhorn's.

Copying @phanle1010 and @james-munson, who I was discussing this with.

**Comment by bk201** [comment]
Thanks @ejweber for the heads-up. We'll evaluate the change.

**Comment by TachunLin** [comment]
Verified **fixed** on `master-6ca6acfd-head` (24/01/23). Close this issue.

### Result
<details><summary>$\color{green}{\textsf{PASS}}$ Case 1: snapshot can work on a stopped VM $~~$</summary>
* VM volumes are detached after we stop VM. 


https://github.com/harvester/harvester/assets/29251855/2e9bb93e-2dc0-432c-8884-f6bfb275e32b

  ```
  node1:~ # kubectl get volume -A
  NAMESPACE         NAME                                       STATE      ROBUSTNESS   SCHEDULED   SIZE          NODE   AGE
  longhorn-system   pvc-5a861225-920d-4059-b501-f02b2fd0ff27   detached   unknown                  10737418240          19m
  ```

* Take the vm snapshot, The snapshot of vm in off state can be ready.
![image](https://github.com/harvester/harvester/assets/29251855/35f3c4ed-e8cb-4492-b99e-98ca91648398)


 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ Case 2: restore a snapshot from detached volumes can work. $~~$</summary>

* Restore the snapshot to a new VM. The new VM can be ready.


https://github.com/harvester/harvester/assets/29251855/fd97f29a-5de9-4d4b-99ea-b771b0a4b716


* Restore the snapshot to replace the old VM. The old VM can be ready.


https://github.com/harvester/harvester/assets/29251855/66b21890-1fed-40b0-942d-f2e42461a1e2



 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ Case 3: backup can work on a stopped VM. $~~$</summary>

* VM volumes are detached after we stop VM. 

  ```
  NAMESPACE         NAME                                       STATE      ROBUSTNESS   SCHEDULED   SIZE          NODE   AGE
  longhorn-system   pvc-d1226d97-ab90-4d40-92f9-960b668093c2   detached   unknown                  10737418240          5m12s
  ```

* Take the vm backup, The backup of vm in off state can be ready.
![image](https://github.com/harvester/harvester/assets/29251855/7d8ba60a-85e4-476f-86ca-1a6ff68b83b1)


 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ Case 4: race condition doesn't break VMBackup $~~$</summary>

* Take multiple backup on the VM in a short time. All backup can be ready.


https://github.com/harvester/harvester/assets/29251855/78246bb6-62ed-4f87-90f1-764ed97c287d


![image](https://github.com/harvester/harvester/assets/29251855/1ea92349-2c5a-4721-a59a-2e83ea3f9724)


 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ Case 5: restore a backup from detached volumes can work $~~$</summary>

* Restore the backup to a new VM. The new VM can be ready.


https://github.com/harvester/harvester/assets/29251855/be80cb02-9b9b-4e7e-9634-3dfad619d39a



* Restore the backup to replace the old VM. (Retain Volume), The old VM can be ready.


https://github.com/harvester/harvester/assets/29251855/7cfac1b1-4f46-4b71-b8cd-8e46f1885af5


* Restore the backup to replace the old VM. (Delete Volume), The old VM can be ready.


https://github.com/harvester/harvester/assets/29251855/c1a4b5ca-e03d-4a66-af77-aece1b0c890d



 </details>



### Test Information
* Test Environment: 3 nodes harvester on baremetal machine
* Harvester version: `master-6ca6acfd-head` (24/01/23).

### Verify Steps
Set the NFS backup target
<details><summary>Case 1: snapshot can work on a stopped VM.</summary>

1. Create a VM.
2. After the VM is ready, stop the VM.
3. Check VM volumes are detached.
4. Take a snapshot on the VM. The snapshot can be ready.

</details>

<details><summary>Case 2: restore a snapshot from detached volumes can work.</summary>

1. Follow Case 1.
2. Make sure VM volumes are detached.
3. Restore the snapshot to a new VM. The new VM can be ready.
4. Restore the snapshot to replace the old VM. The old VM can be ready.

</details>

<details><summary>Case 3: backup can work on a stopped VM</summary>

1. Create a VM.
2. After the VM is ready, stop the VM.
3. Check VM volumes are detached.
4. Take a backup on the VM. The backup can be ready.

</details>

<details><summary> Case 4: race condition doesn't break VMBackup </summary>

1. Create a VM.
2. After the VM is ready, stop the VM.
3. Check VM volumes are detached.
4. Take multiple backup on the VM in a short time. All backup can be ready.

</details>

<details><summary>Case 5: restore a backup from detached volumes can work</summary>

1. Follow Case 3.
2. Make sure VM volumes are detached.
3. Restore the backup to a new VM. The new VM can be ready.
4. Restore the backup to replace the old VM. The old VM can be ready.

</details>
